src/popup/modal.js

```diff
@@ -0,0 +1,67 @@
+// import './api.js'
+import getCharacters from '../modules/fetchAPI.js';
+
+const modalContainer = document.querySelector('.popup-window');
+const overlay = document.querySelector('.overlay');
+
+const url = 'https://rickandmortyapi.com/api/character';
+
+const displayModal = (id) => {
+  getCharacters(url)
+    .then((request) => {
+      const charactersArray = request.results;
+      charactersArray.forEach((character) => {
+        //  for (const character of charactersArray) {
+        if (Number(id) === character.id) {
+          modalContainer.style.display = 'flex';
+          overlay.style.display = 'block';
+          let html = '';
+          html += `
+        <div class= "title-modal">
+            <i class="fas fa-times"></i>
+        </div>
+        <div class="image">
+            <img src="${character.image}" alt="image">
+            <div class="name"> ${character.name} </div>
+        </div>
+        <div class="details">
+            <div class="box"> Status : ${character.status} </div>
+            <div class="box"> Species : ${character.species} </div>
+            <div class="box"> Type : ${character.type} </div>
+            <div class="box"> Gender : ${character.gender} </div>
+        </div>
+        <div class="comment-box">
+          <h2 class="title"> Comments (2) </h2>
+          <div class="comment-container">
+            <p> 15/03/2023 John: I'd love to buy it so much </p>
+            <p> 15/03/2023 John: I'd love to buy it </p>
+          </div>
+        </div>
```

Comment:
I kindly recommend you to check the requirements again, you are showing the comments section, that great but the comments of that section should be from the data from the API of this endpoint /apps/:app_id/comments, you can add some comments using the following structure in the body of the POST request: 

{
    "item_id": "item1",
    "username": "Jane",
    "comment": "Hello"
}

You can use the involvement API class and add a method to get comments, remember that the post comment method is not a requirement for this activity you'll do it on an upcoming activity

You can take a look at the get likes method to create the get comments method they suppose to be alike